ClangTidy warnings for `modernize-use-equals-default`

```
/Users/jviotti/Projects/core/src/core/json/include/sourcemeta/core/json_array.h:15:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
   15 |   JSONArray() : data{} {}
      |   ^                    ~~
      |                        = default;

/Users/jviotti/Projects/core/src/core/jsonl/iterator.cc:107:21: error: use '= default' to define a trivial destructor [modernize-use-equals-default,-warnings-as-errors]
  107 | ConstJSONLIterator::~ConstJSONLIterator() {}
      |                     ^                     ~~
      |                                           = default;

/Users/jviotti/Projects/core/src/core/jsonpointer/include/sourcemeta/core/jsonpointer_pointer.h:34:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
   34 |   GenericPointer() : data{} {}
      |   ^                         ~~
      |                             = default;

/Users/jviotti/Projects/core/src/core/jsonpointer/include/sourcemeta/core/jsonpointer_template.h:41:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
   41 |   GenericPointerTemplate() : data{} {}
      |   ^                                 ~~
      |                                     = default;

/Users/jviotti/Projects/core/src/core/jsonschema/resolver.cc:10:20: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
   10 | SchemaMapResolver::SchemaMapResolver() {}
      |                    ^                   ~~
      |                                        = default;

/Users/jviotti/Projects/core/src/core/jsonschema/resolver.cc:87:25: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
   87 | SchemaFlatFileResolver::SchemaFlatFileResolver() {}
      |                         ^                        ~~
      |                                                  = default;
```

Same in all of the AlterSchema rules too. Here is a summary form, as otherwise the text is too long for GitHub to handle:

```
/Users/jviotti/Projects/core/src/extension/alterschema/antipattern/const_with_type.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/antipattern/duplicate_enum_values.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/antipattern/duplicate_required_values.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/antipattern/enum_with_type.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/antipattern/exclusive_maximum_number_and_maximum.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/antipattern/exclusive_minimum_number_and_minimum.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/desugar/boolean_true.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/desugar/const_as_enum.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/desugar/exclusive_maximum_integer_to_maximum.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/desugar/exclusive_minimum_integer_to_minimum.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/desugar/type_array_to_any_of_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/desugar/type_boolean_as_enum.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/desugar/type_null_as_enum.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/implicit/max_contains_covered_by_max_items.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/implicit/min_items_given_min_contains.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/implicit/min_items_implicit.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/implicit/min_length_implicit.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/implicit/min_properties_covered_by_required.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/implicit/min_properties_implicit.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/implicit/multiple_of_implicit.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/implicit/properties_implicit.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/implicit/type_union_implicit.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/additional_properties_default.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/content_schema_default.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/dependencies_default.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/dependent_required_default.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/items_array_default.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/items_schema_default.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/pattern_properties_default.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/properties_default.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/unevaluated_items_default.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/unevaluated_properties_default.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/unsatisfiable_max_contains.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/redundant/unsatisfiable_min_properties.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/simplify/dependencies_property_tautology.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/simplify/dependent_required_tautology.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/simplify/equal_numeric_bounds_to_enum.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/simplify/maximum_real_for_integer.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/simplify/minimum_real_for_integer.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/simplify/single_type_array.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/content_media_type_without_encoding.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/content_schema_without_media_type.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_applicator_2019_09.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_applicator_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_content_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_content_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_draft0.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_draft1.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_draft2.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_draft3.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_draft4.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_draft6.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_draft7.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_format_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_format_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_unevaluated_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_validation_2019_09.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_array_keywords_validation_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_applicator_2019_09.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_applicator_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_content_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_content_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_draft0.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_draft1.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_draft2.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_draft3.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_draft4.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_draft6.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_draft7.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_format_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_format_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_unevaluated_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_validation_2019_09.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_boolean_keywords_validation_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_applicator_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_applicator_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_content_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_content_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_draft0.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_draft1.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_draft2.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_draft3.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_draft4.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_draft6.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_draft7.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_format_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_format_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_unevaluated_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_validation_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_null_keywords_validation_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_applicator_2019_09.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_applicator_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_content_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_content_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_draft0.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_draft1.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_draft2.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_draft3.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_draft4.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_draft6.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_draft7.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_format_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_format_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_unevaluated_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_validation_2019_09.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_numeric_keywords_validation_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_applicator_2019_09.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_applicator_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_content_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_content_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_draft0.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_draft1.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_draft2.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_draft3.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_draft4.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_draft6.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_draft7.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_format_2019_09.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_format_2020_12.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_unevaluated_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_validation_2019_09.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_object_keywords_validation_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_applicator_2019_09.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_applicator_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_draft0.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_draft1.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_draft2.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_draft3.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_draft4.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_draft6.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_draft7.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_unevaluated_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_validation_2019_09.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/drop_non_string_keywords_validation_2020_12.h:4:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/duplicate_allof_branches.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/duplicate_anyof_branches.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/else_without_if.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/if_without_then_else.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/max_contains_without_contains.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/min_contains_without_contains.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/superfluous/then_without_if.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
/Users/jviotti/Projects/core/src/extension/alterschema/syntax_sugar/enum_to_const.h:3:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
```

--- a/src/core/jsonl/iterator.cc
+++ b/src/core/jsonl/iterator.cc
@@ -104,7 +104,7 @@ ConstJSONLIterator::ConstJSONLIterator(
     std::basic_istream<JSON::Char, JSON::CharTraits> *stream)
     : data{stream}, internal{new Internal({this->parse_next()})} {}
 
-ConstJSONLIterator::~ConstJSONLIterator() {}
+ConstJSONLIterator::~ConstJSONLIterator() = default;
 
 auto operator==(const ConstJSONLIterator &left, const ConstJSONLIterator &right)
     -> bool {

--- a/src/core/jsonschema/resolver.cc
+++ b/src/core/jsonschema/resolver.cc
@@ -5,7 +5,7 @@
 
 namespace sourcemeta::core {
 
-SchemaMapResolver::SchemaMapResolver() {}
+SchemaMapResolver::SchemaMapResolver() = default;
 
 SchemaMapResolver::SchemaMapResolver(const SchemaResolver &resolver)
     : default_resolver{resolver} {}

